### ISSUE TITLE
g10k assumes cache directory is populated if it exists

### ISSUE THREAD

**Issue by andrewfraley** [xorpaul/g10k#65]
If the cache directory exists, but is empty, g10k tries to update the non-existent repo instead of cloning it.  Not a show stopping bug, but it took a bit to figure out why it kept telling me my control repo was unreachable.  

Thanks for this project btw, our r10k runs were approaching 3 minutes and g10k does it in 3 seconds with only 25 workers!

```
---
:cachedir: '/var/cache/g10k'
sources:
  example:
    remote: 'https://github.localdomain/myorg/r10k.git'
    basedir: '/etc/puppetlabs/code/environments'
```

> [root@puppetmaster ~]# mkdir /var/cache/g10k
> [root@puppetmaster ~]# chown puppet:puppet /var/cache/g10k
> [root@puppetmaster ~]# su - puppet
> [puppet@puppetmaster ~]$ /opt/puppetlabs/puppet/bin/g10k -config /etc/puppetlabs/g10k/g10k_config.yaml
> **WARN: git repository https://github.localdomain/myorg/r10k.git does not exist or is unreachable at this moment!**

**Comment by andrewfraley** [comment]
Ah sorry PEBKAC, I kept creating /var/cache/g10k as a file not a directory, DERP.

**Comment by xorpaul** [comment]
I was wondering why you used `touch` :)

I'm still going to fix g10k to check if the directories are really directories and not just files.

**Comment by xorpaul** [comment]
Fixed with v0.3.8 https://github.com/xorpaul/g10k/releases/tag/v0.3.8 https://github.com/xorpaul/g10k/commit/58958e9a1fba7ede7bac563e081a1fb901db0c32
